contracts/registry/src/lib.rs

```diff
@@ -170,12 +171,12 @@ impl Contract {
                 emit_soul_transfer(&owner, &recipient);
             }
         } else {
-            let last = &batch[token_counter];
+            let last = &batch[token_counter - 1];
             self.ongoing_soul_tx.insert(
                 &owner,
                 &IssuerTokenId {
                     issuer_id: last.0.issuer_id,
-                    token: last.1,
+                    token: last.0.class_id,
```

Comment:
```suggestion
                    token: last.0.class_id,  // we reise IssuerTokenId type here (to not generate new code), but we store class_id instead of token here.
```